website-custom-views/src/content/development/testing.mdx

```diff
@@ -0,0 +1,321 @@
+
+The main function you would want to use from that package is named `renderCustomView`.
+
+We recommend writing a function, for example `renderView`, to encapsulate the basic test setup, so that your actual test remains as clean as possible.
+
+```js title="channels.spec.js"
+import {
+  fireEvent,
+  screen,
+  renderCustomView,
+} from '@commercetools-frontend/application-shell/test-utils';
+import { entryPointUriPath } from '../../constants/application';
+import ApplicationRoutes from '../../routes';
+
+const renderView = (options = {}) => {
+  renderCustomView({
+    locale: options.locale || 'en',
+    projectKey: options.projectKey || 'my-project',
+    children: <ApplicationRoutes />,
+  });
+};
+
+it('should render channels', async () => {
+  mockServer.use(/* See mock setup */);
+
+  renderView();
+
+  await screen.findByText('Channel no. 0');
+});
+```
+
+With this simple test we implicitly have tested the following things:
+
+* The routes work.
+* The channels page renders.
+* The data is fetched.
+* The data is displayed on the page.
+
+From here you can enhance the test to do other things, especially testing the user interactions. For example:
+
+* We can simulate that the user paginates through the table, or use search and filters.
+
+Let's enhance our test to paginate to page two. First we need to adjust our GraphQL mock to return results based on the `offset` from the query variables. Then we need to interact with the pagination button to go to the next page.
+
+```js title="channels.spec.js" highlightLines="1,5-7,14,30-31,34,37-38"
+it('should render channels and paginate to second page', async () => {
+  mockServer.use(
+    graphql.query('FetchChannels', (req, res, ctx) => {
+      // Simulate a server side pagination.
+      const { offset } = req.variables;
+      const totalItems = 25; // 2 pages
+      const itemsPerPage = offset === 0 ? 20 : 5;
+
+      return res(
+        ctx.data({
+          channels: buildGraphqlList(
+            Array.from({ length: itemsPerPage }).map((_, index) =>
+              ChannelMock.random().key(
+                `channel-key-${offset === 0 ? index : 20 + index}`
+              )
+            ),
+            {
+              name: 'Channel',
+              total: totalItems,
+            }
+          ),
+        })
+      );
+    })
+  );
+
+  renderView();
+
+  // First page
+  await screen.findByText('Channel no. 0');
+  expect(screen.queryByText('Channel no. 22')).not.toBeInTheDocument();
+
+  // Go to second page
+  fireEvent.click(screen.getByLabelText('Next page'));
+
+  // Second page
+  await screen.findByText('Channel no. 22');
+  expect(screen.queryByText('Channel no. 0')).not.toBeInTheDocument();
+});
+```
+
+This should give you a basic idea on how you can approach testing your Custom View.
+
+## Testing user permissions
```

Comment:
Should we include this section? 
We haven't even testing how e2e would look like for Custom Views.